test/components/lenra_form_test.dart

```diff
@@ -0,0 +1,96 @@
+import 'package:flutter/material.dart';
+import 'package:lenra_components/component/lenra_checkbox.dart';
+import 'package:lenra_components/component/lenra_toggle.dart';
+import 'package:lenra_ui_runner/components/events/data/value_data.dart';
+import 'package:lenra_ui_runner/components/events/event.dart';
+import 'package:lenra_ui_runner/lenra_widget.dart';
+import 'package:lenra_ui_runner/widget_model.dart';
+import 'package:provider/provider.dart';
+import "../test_helper.dart";
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  testWidgets('LenraForm should work properly', (WidgetTester tester) async {
```

Comment:
You should simplify your unit test and add A LOT more of them.
This will help you to catch all the small errors in your code !

For each input, you probably should have at least 3 tests : 
- One "nominal" test. Add the input alone in a form, add a name, the submit button, click the button and check that the event return the correct value. Update the input value, click the submit again, check that the value in the event actually changed.
- One test where you add the input outside of a form and add a name. Check that the input does not crash if you use it.
- One test where you add the input inside a form but you don't add a name. Check that the event does NOT send the value of this input.
- One nominal test but you also add a listener in the input.

Add some tests for the button : 
- one test that check if the button correctly trigger the form event if the submit is true.
- One test that check it does not crash if the submit is true but outside of a form.
- One test that check it does not send the form event if submit is false and/or not specified (default to false)
- One test that check that two event are actually sent if the button is on a form with submit = true AND it does have a listener too.

For the form tests : 
- Add a test that check if nestead forms actually work as expected.